pkg/plugin/GlobalPluginService.go

```diff
@@ -1805,24 +1750,62 @@ func (impl *GlobalPluginServiceImpl) ListAllPluginsV2(filter *bean2.PluginsListF
 
 	return pluginDetails, nil
 }
+func (impl *GlobalPluginServiceImpl) validateDetailRequest(pluginVersions []*repository.PluginMetadata, pluginVersionIds, parentPluginIds []int) error {
+	pluginVersionsIdMap, pluginParentIdMap := make(map[int]bool, len(pluginVersionIds)), make(map[int]bool, len(parentPluginIds))
+	allPlugins, err := impl.globalPluginRepository.GetAllPluginMinData()
+	if err != nil {
+		impl.logger.Errorw("validateDetailRequest, error in getting all plugins parent metadata", "err", err)
+		return err
+	}
+	for _, pluginVersion := range pluginVersions {
+		pluginVersionsIdMap[pluginVersion.Id] = true
+	}
+	for _, plugin := range allPlugins {
+		pluginParentIdMap[plugin.Id] = true
+	}
+	nonExistingPluginVersionIds, nonExistingPluginParentIds := make([]int, 0, len(pluginVersionIds)), make([]int, 0, len(parentPluginIds))
+	for _, versionId := range pluginVersionIds {
+		if _, ok := pluginVersionsIdMap[versionId]; !ok {
+			nonExistingPluginVersionIds = append(nonExistingPluginVersionIds, versionId)
+		}
+	}
+	for _, pluginId := range parentPluginIds {
+		if _, ok := pluginParentIdMap[pluginId]; !ok {
+			nonExistingPluginParentIds = append(nonExistingPluginParentIds, pluginId)
```

Comment:
instead of storing this in array and checking later, use an errorMessage and set it if !ok. Break the loop , this will save us iterations and array mem

Reply:
right, point was to show which all versionIds or parentIds are non existing in a single error message